tests/core/test_proxy.py

```diff
@@ -44,3 +65,63 @@ def test_no_proxy(self):
         env.pop("https_proxy", None)
         env.pop("HTTP_PROXY", None)
         env.pop("HTTPS_PROXY", None)
+
+
+class CustomAgentTransaction(AgentTransaction):
+
+    def on_response(self, response):
+        super(CustomAgentTransaction, self).on_response(response)
+        self._test.stop()
+
+class TestProxy(AsyncTestCase):
+    @attr(requires='core_integration')
+    def test_proxy(self):
+        config = {
+            "endpoints": {"https://app.datadoghq.com": ["foo"]},
+            "proxy_settings": {
+                "host": "localhost",
+                "port": PROXY_PORT,
+                "user": None,
+                "password": None
+            }
+        }
+
+        app = Application()
+        app.skip_ssl_validation = True
+        app._agentConfig = config
+
+        trManager = TransactionManager(MAX_WAIT_FOR_REPLAY, MAX_QUEUE_SIZE, THROTTLING_DELAY)
+        trManager._flush_without_ioloop = True  # Use blocking API to emulate tornado ioloop
```

Comment:
Well if I don't set this to True, the transaction is never flushed. I think it is because the IOLoop is started only when I call `wait()`, which I do after creating the transaction.